resin_release_tool/releaser.py

```diff
@@ -21,6 +24,68 @@ def get_canaries(self):
             if tag['tag_key'] == 'CANARY']
         return canaries
 
+    def get_tags_per_device(self):
+        all_apps = self.models.application.get_all()
+        if len(next((app for app in all_apps if app['id'] == int(self.app_id)),{})) != 0: #checks if the app exists
+            print("Found app with app_id ", self.app_id, "in Balena")
+            app_info = next((app for app in all_apps if app['id'] == int(self.app_id)),{})
+            all_devices = self.models.device.get_all_by_application_id(app_info['id'])
+            tags_per_device = [{'uuid':device['uuid'],
+                                'tags':self.models.tag.device.get_all_by_device(device['uuid'])}
+                               for device in all_devices]
+            entries_to_remove = ('id', 'device', '__metadata')
+            for device in tags_per_device: #removes all the key/values except, "tag_key" and "value"
+                for tag in device['tags']:
+                    for k in entries_to_remove:
+                        tag.pop(k)
```

Comment:
This loop seems redundant if you're only taking tag_key and value in the loop below, then deleting/overwriting the whole device['tags']